services/paymaster/src/rpc.rs

```diff
@@ -39,29 +41,58 @@ fn to_error_response(err: Error) -> ErrorResponse {
 }
 
 // Inspired by send_and_confirm_transaction from solana-rpc-client, but accepts a config
+#[tracing::instrument(
+    skip_all,
+    fields(tx_hash = %transaction.signatures[0])
+)]
 pub async fn send_and_confirm_transaction(
     rpc: &RpcClient,
     transaction: &VersionedTransaction,
     config: RpcSendTransactionConfig,
 ) -> Result<ConfirmationResult, ErrorResponse> {
     let recent_blockhash = transaction.message.recent_blockhash();
-    let signature = transaction.signatures[0];
-    match rpc.send_transaction_with_config(transaction, config) {
-        Ok(signature) => signature,
+    let signature = match send_transaction(rpc, transaction, config) {
+        Ok(sig) => sig,
         Err(err) => {
             if let Some(error) = err.get_transaction_error() {
                 return Ok(ConfirmationResult::Failed {
-                    signature: signature.to_string(),
+                    signature: transaction.signatures[0].to_string(),
                     error,
                 });
             }
             return Err(to_error_response(err));
         }
     };
 
+    confirm_transaction(rpc, &signature, recent_blockhash).await
+}
+
+// Wrapper for send_transaction_with_config to add tracing
```

Comment:
i don't love the extremely shallow interface here, but it was done to keep all the tracing in the macros. we could also get rid of this and just directly trace in the non-confirmation path with

```
tracing::info_span!("send_transaction", tx_hash = %transaction.signatures[0])
```

but i slightly preferred separating out all the tracing into macros. open to feedback